src/com/google/enterprise/adaptor/sharepoint/SharePointAdaptor.java

```diff
@@ -3411,6 +3379,67 @@ private HtmlResponseWriter createHtmlResponseWriter(
     public SiteDataClient getSiteDataClient() {
       return siteDataClient;
     }
+      
+    /** 
+     * MemberIdMapping will have information about users with explicit
+     * permissions on SharePoint or users which are direct members of
+     * SharePoint groups. MemberIdMapping might not have information
+     * about all valid SharePoint Users. To get all valid SharePoint users
+     * under SiteCollection, use SiteUserMapping. 
+     */
+    private class IdMappings {
+      private MemberIdMapping memberIdMapping;
+      private MemberIdMapping siteUserMapping;
+      private boolean memberIdMappingRefreshed = false;
+      private boolean siteUserMappingRefreshed = false;
+      private boolean siteUserMappingInitialized = false;
+
+      private IdMappings() throws IOException {
+        this.memberIdMapping = getMemberIdMapping();
+      }
+
+      private Principal resolvePrincipal(Integer id) throws IOException {
+        Principal principal = memberIdMapping.getPrincipal(id);
+        if (principal == null) {
+          log.log(Level.FINE, "Member id {0} is not available in memberid"
+              + " mapping for Web [{1}] under Site Collection [{2}].",
+              new Object[] {id, webUrl, siteUrl});
+          if (siteUserMapping == null && !siteUserMappingInitialized) {
+            try {
+              siteUserMapping = getSiteUserMapping();
+              principal = siteUserMapping.getPrincipal(id);
```

Comment:
This call should be removed, since we will now fall through to the one farther down.